es/awsClientSigner.go

```diff
@@ -50,16 +50,16 @@ func checkParametersError(endpointMetaData []string, creds *credentials.Credenti
 //		- endpoint: The endpoint URI gettable from AWS.
 //		- creds: Credentials from AWS/Credentials.
 func NewSignedElasticClient(endpoint string, creds *credentials.Credentials) (*elastic.Client, error) {
-	if cofs, err := NewSignedElasticClientOptions(endpoint, creds); err == nil {
-		cof := configEach(cofs...)
-		if ec, err := elastic.NewClient(elastic.SetURL(endpoint), cof); err == nil {
-			return ec, nil
-		} else {
-			return nil, err
-		}
-	} else {
+	cofs, err := NewSignedElasticClientOptions(endpoint, creds)
+	if err != nil {
 		return nil, err
 	}
+	cof := configEach(cofs...)
+	ec, err := elastic.NewClient(elastic.SetURL(endpoint), cof)
```

Comment:
This function has no real problem with variable shadowing and you also do it on this modification

Reply:
I also wanted to remove as many occurrences of shadowing, however innocent they might actually be, as possible, so that finding future problems with usage of shadowing would be simpler, but if you want I can revert this change.

Reply:
Both versions are correct but the logic is reversed. What do you think @simonmeyerrr ?

Reply:
(also, I cannot seem to find any formally written traces of this but I believe using `if err == nil` is considered an anti-pattern in general by the Go devs)

Reply:
(Found it, [here](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow))